### DISCUSSION TITLE
Can't seem to get code lenses to work with gopls

### DISCUSSION THREAD

**Discussion by jtl5770** [emacs-lsp/lsp-mode#3121] [Q&A] [answered]
Hi,

I am using up-to date lsp and friends from melpa. WIth my setup, I will happily see code lenses when doing java code, but nothing ever shows up for go. I have all variables enabled that have something to do with code-lenses (also the go specific ones) still nothing can be seen.

Looking at the lsp severs capabilities, it simply states "codeLensProvider: " - no "t" oder anything else after that.
Can anybody enlighten me what may be wrong here? @yyoncho mention that @muirdm may be interested/of help here.

**Comment by jtl5770** [comment]
interestingly - when I enable lsp-mode for go.mod files, inside those it shows me a couple of code lenses.
Still I have no idea if this is what SHOULD happen, and there simply are no other code lenses defined for "normal" go files, or if this is a bug somewhere.

**Comment by muirdm** [comment, marked as answer]
There aren't very many code lenses. [Here](https://github.com/golang/tools/blob/master/gopls/doc/settings.md#codelenses-mapstringbool) you can see most of the lenses are for go.mod files. For regular files you can opt in to "gc_details", and there is one for go generate commands and cgo. Also, there is a code lens called "test" you can opt in to that lets you run individual tests (not sure why it isn't documented).

**Reply by jtl5770** [reply]
Ok, thanks for the information... I was kind of hoping to have similar lenses as in java (references, definition or the like). But at least I didn't miss any. I was already suspecting that when I found out that they are shown in the go.mod files.
